batch/batch/aioclient.py

```diff
@@ -0,0 +1,328 @@
+import math
+import time
+import random
+import yaml
+import cerberus
+import aiohttp
+
+from . import schemas
+
+
+class Job:
+    def __init__(self, client, id, attributes=None, parent_ids=None, scratch_folder=None, _status=None):
+        if parent_ids is None:
+            parent_ids = []
+        if attributes is None:
+            attributes = {}
+
+        self.client = client
+        self.id = id
+        self.attributes = attributes
+        self.parent_ids = parent_ids
+        self.scratch_folder = scratch_folder
+        self._status = _status
+
+    async def is_complete(self):
+        if self._status:
+            state = self._status['state']
```

Comment:
delete `if self._status: ...`

Reply:
Why?  `_status` can be None if it is unknown.

Reply:
I thought we'd always want to call self.status() which updates self._status. Is `self._status` supposed to be the cached version? What if it is out of date?